app/routes/__index/domains/index.tsx

```diff
@@ -16,13 +22,51 @@ export const loader = async ({ request }: LoaderArgs) => {
 };
 
 export const action = async ({ request }: ActionArgs) => {
-  if (request.method === 'PUT') {
-    // TODO: Call the appropriate code to prolong record
-  } else if (request.method === 'DELETE') {
-    // TODO: Call the appropriate code to delete record
+  const username = await requireUsername(request);
+  const formData = await request.formData();
+  const intent = formData.get('intent');
+  const id = formData.get('id');
+  invariant(intent, 'Missing `intent` in posted form data');
+  invariant(id, 'Missing `id` in posted form data');
+
+  const recordID = Number(id);
+  const record = await getRecordById(recordID);
+
+  if (!record) {
+    throw new Response('The record is not found', {
+      status: 422,
+    });
   }
 
-  return null;
+  switch (intent) {
+    case 'renew-record':
+      await updateDnsRequest({
```

Comment:
Question: is a "renew" trying to update the expiry date?  It doesn't look like we actually do that in the db: https://github.com/DevelopingSpace/starchart/blob/3fa7f7079b2df10aad33ef3d96450c3fe9dc9c37/app/queues/dns/dns-flow.server.ts#L87-L107. cc @Genne23v

Reply:
`updateRecordStatusById` also updates the expiry date to the date now + 6 months

Reply:
In that case, you can add this function to `record.server.ts`. 
```js
export async function renewRecordById(id: Record['id']) {
  const expiresAt = dayjs().set('month', 6).toDate();
  return prisma.record.update({
    where: {
      id,
    },
    data: {
      expiresAt,
    },
  });
}
```